drivers/clock_control/clock_stm32_ll_u5.c

```diff
@@ -386,149 +281,170 @@ void config_src_sysclk_hse(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
-{
-#ifdef STM32_SYSCLK_SRC_MSIS
+		LL_PWR_DisableBkUpAccess();
+	}
 
-	uint32_t old_hclk_freq;
-	uint32_t new_hclk_freq;
+	if (IS_ENABLED(STM32_MSIS_ENABLED)) {
+		/* Set MSIS Range */
+		LL_RCC_MSI_EnableRangeSelection();
 
-	old_hclk_freq = HAL_RCC_GetHCLKFreq();
+		LL_RCC_MSIS_SetRange(STM32_MSIS_RANGE << RCC_ICSCR1_MSISRANGE_Pos);
 
-	/* Calculate new SystemCoreClock variable with MSI freq */
-	/* MSI freq is defined from RUN range selection */
-	new_hclk_freq =	__LL_RCC_CALC_HCLK_FREQ(
-				__LL_RCC_CALC_MSIS_FREQ(LL_RCC_MSIRANGESEL_RUN,
-				STM32_MSIS_RANGE << RCC_ICSCR1_MSISRANGE_Pos),
-				s_ClkInitStruct.AHBCLKDivider);
+		if (IS_ENABLED(STM32_MSIS_PLL_MODE)) {
+			__ASSERT(STM32_LSE_ENABLED,
+				"MSIS Hardware auto calibration needs LSE clock activation");
+			/* Enable MSI hardware auto calibration */
+			LL_RCC_MSI_EnablePLLMode();
+		}
 
-	__ASSERT(new_hclk_freq == CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC,
-			 "Config mismatch HCLK frequency %u %u",
-			 CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC, new_hclk_freq);
+		/* Enable MSIS */
+		LL_RCC_MSIS_Enable();
 
-	/* If freq increases, set flash latency before any clock setting */
-	if (new_hclk_freq > old_hclk_freq) {
-		LL_SetFlashLatency(new_hclk_freq);
+		/* Wait till MSIS is ready */
+		while (LL_RCC_MSIS_IsReady() != 1) {
+		}
 	}
 
-	set_regu_voltage(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
+	if (IS_ENABLED(STM32_MSIK_ENABLED)) {
+		/* Set MSIS Range */
+		LL_RCC_MSI_EnableRangeSelection();
 
-	/* Set MSIS as SYSCLCK source */
-	set_up_clk_msis();
-	LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_MSIS);
-	LL_RCC_SetAHBPrescaler(s_ClkInitStruct.AHBCLKDivider);
-	while (LL_RCC_GetSysClkSource() != LL_RCC_SYS_CLKSOURCE_STATUS_MSIS) {
-	}
+		LL_RCC_MSIK_SetRange(STM32_MSIK_RANGE << RCC_ICSCR1_MSIKRANGE_Pos);
 
-	/* Update SystemCoreClock variable */
-	LL_SetSystemCoreClock(new_hclk_freq);
+		if (IS_ENABLED(STM32_MSIK_PLL_MODE)) {
+			__ASSERT(STM32_LSE_ENABLED,
+				"MSIK Hardware auto calibration needs LSE clock activation");
+			/* Enable MSI hardware auto calibration */
+			LL_RCC_MSI_EnablePLLMode();
+		}
 
-	/* Set peripheral busses prescalers */
-	LL_RCC_SetAPB1Prescaler(s_ClkInitStruct.APB1CLKDivider);
-	LL_RCC_SetAPB2Prescaler(s_ClkInitStruct.APB2CLKDivider);
-	LL_RCC_SetAPB3Prescaler(s_ClkInitStruct.APB3CLKDivider);
+		if (IS_ENABLED(STM32_MSIS_ENABLED)) {
+			__ASSERT((STM32_MSIK_PLL_MODE == STM32_MSIS_PLL_MODE),
+				"Please check MSIS/MSIK config consistency");
+		}
 
-	/* If freq not increased, set flash latency after all clock setting */
-	if (new_hclk_freq <= old_hclk_freq) {
-		LL_SetFlashLatency(new_hclk_freq);
+		/* Enable MSIK */
+		LL_RCC_MSIK_Enable();
+
+		/* Wait till MSIK is ready */
+		while (LL_RCC_MSIK_IsReady() != 1) {
+		}
 	}
 
-	/* Disable other clocks */
-	LL_RCC_HSE_Disable();
-	LL_RCC_HSI_Disable();
-	LL_RCC_PLL1_Disable();
+	if (IS_ENABLED(STM32_LSI_ENABLED)) {
+		/* Enable LSI oscillator */
+		LL_RCC_LSI_Enable();
+		while (LL_RCC_LSI_IsReady() != 1) {
+		}
+	}
```

Comment:
open for further update to include the LSI prediv  on the stm32u5 ( LL_RCC_LSI_SetPrescaler)